pyramda/prop.py

```diff
@@ -0,0 +1,8 @@
+from pyramda.function.curry import curry
+
+
+@curry
+def prop(x, obj):
+    if isinstance(obj, dict):
```

Comment:
I like when I get to think about the fundamentals of a language for a bit. Thanks for taking the time to. Though about it for a moment and your reasoning is certainly more respectful towards Python itself, which is obviously a good idea.

So, I did notice there is already a `getattr` implementation in there and also a `getitem` in the `dictionary` folder, so I will remove the prop thing entirely :)